### PR TITLE
Add docs for startup strategy information/examples (#1157)

### PR DESCRIPTION
**Pull request by vcvitaly**
Add a section 'Startup Strategies' to docs/features/startup_and_waits.md describing the default and 'one-shot' strategies, with a codeincluded example for the latter.

### CODE REVIEW

**Comment by rnorth** [comment]
Thanks @vcvitaly - I'll have a read tomorrow.
Please ignore the CircleCI failures - that is a test reporting issue that we need to fix at our end 😬

**Comment by bsideup** [comment]
@rnorth ping

File: docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    }
+
+    public static class IndefiniteOneShotStrategyTest {
+        @Rule
+        // withIndefiniteOneShotStrategy {
+        public GenericContainer bboxWithIndefiniteOneShot = new GenericContainer("busybox:1.31.1")
+            .withCommand("sh", "-c", String.format("sleep 5 && echo \"%s\"", HELLO_TESTCONTAINERS))
+            .withStartupCheckStrategy(
+                new IndefiniteWaitOneShotStartupCheckStrategy().withTimeout(Duration.ofSeconds(3))
```
Comment (rnorth) [inline docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java:59]: It does seem slightly odd to use a timeout with this strategy - is it intentional?

I'd suggest just:
```suggestion
                new IndefiniteWaitOneShotStartupCheckStrategy()
```
should be appropriate...

Comment (rnorth) [reply docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java:59]: Will apply and merge.

Comment (vcvitaly) [reply docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java]: @rnorth I'm sorry for not replying earlier, I was a little busy, but once I saw that you are going to commit I decided to reply straight away :)

That was intentional to show that even though timeout is 3 sec, the command executed on startup won't be interrupted and will continue executing as long as required (which is 5 sec)

**Comment by rnorth** [comment]
Merged, thanks @vcvitaly!

**Pull request merged**